packages/core/strapi/lib/services/entity-service/types/params/publication-state.ts

```diff
@@ -0,0 +1,30 @@
+import type { Common, Utils } from '@strapi/strapi';
+
+export type Kind = 'preview' | 'live';
+
+export type IsEnabled<TSchemaUID extends Common.UID.Schema> = Utils.Expression.MatchFirst<
+  [
+    [
+      Common.UID.IsContentType<TSchemaUID>,
+      Utils.Expression.IsTrue<Common.Schemas[TSchemaUID]['options']['draftAndPublish']>
+    ],
+    [Common.UID.IsComponent<TSchemaUID>, Utils.Expression.False]
+  ],
+  Utils.Expression.BooleanValue
+>;
```

Comment:
I think I understand what the MatchFirst does, because I understand when D&P is enabled.
But I wouldnt be able to use it now.

How does it work? 🤔

Reply:
yes :)

Reply:
(if yes)

It basically loops over the given test cases (`Array<[boolean, unknown]>`), if the boolean value is `true`, then it returns the associated value. Else, it'll go to the next item in the test array.

If at the end of the loop, it didn't find any result, then the type will default to the guard/default value (second parameter)

Reply:
You can find the implementation here: https://github.com/strapi/strapi/blob/main/packages/core/strapi/lib/types/utils/expression.d.ts#L17C1-L28C11

Reply:
I see! thank you that was very clear :) 
I think it could be useful to add this info in the Utils.Expression.Matchfirst as comment!

Reply:
Definitely, something I plan to do (adding doc for all the utils). Thanks for the reminder! :slightly_smiling_face: